### PR TITLE
survive app restart by ignoring already_running

### PR DESCRIPTION
**Pull request by DEvil0000**
ticket #25

### CODE REVIEW

**Comment by id** [comment]
I think the right way to solve this would be to do `application:stop(cowboy)` in `brucke_app:stop/1`

**Comment by DEvil0000** [comment]
Some of my other applications are also using cowboy so i do not want to stop it.


Von meinem Samsung Gerät gesendet.

-------- Ursprüngliche Nachricht --------
Von: Ivan Dyachkov <notifications@github.com> 
Datum: 07.07.2017  09:00  (GMT+01:00) 
An: klarna/brucke <brucke@noreply.github.com> 
Cc: "A. Binzxxxxxx" <alexander@binzberger.de>, Author <author@noreply.github.com> 
Betreff: Re: [klarna/brucke] survive app restart by ignoring already_running
  (#27) 

I think the right way to solve this would be to do application:stop(cowboy) in brucke_app:stop/1

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/klarna/brucke","title":"klarna/brucke","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/klarna/brucke"}},"updates":{"snippets":[{"icon":"PERSON","message":"@id in #27: I think the right way to solve this would be to do `application:stop(cowboy)` in `brucke_app:stop/1`"}],"action":{"name":"View Pull Request","url":"https://github.com/klarna/brucke/pull/27#issuecomment-313603686"}}}

File: src/brucke_http.erl [hunk 1]
```diff
@@ -26,11 +26,17 @@ init() ->
...
-
+  case cowboy:start_http(http, 8, [{port, Port}],
+                                  [ {env, [{dispatch, Dispatch}]}
+                                  , {onresponse, fun error_hook/4}]) of
+    {ok, _Pid} ->
+      ok;
+    {error, {already_started, _Pid}} ->
+      ok;
+    _else ->
+      error
```
Comment (id) [inline src/brucke_http.erl]: `s/_else/Other` and return `{error, Other}`

File: src/brucke_metrics.erl [hunk 2]
```diff
@@ -45,8 +45,14 @@ init() ->
...
                Port -> [{port, Port} | Opts0]
              end,
-      {ok, _Pid} = graphiter:start(?WRITER, Opts),
-      ok
+      case graphiter:start(?WRITER, Opts) of
+        {ok, _Pid} ->
+          ok;
+        {error, {already_started, _Pid}} ->
+          ok;
+        _else ->
```
Comment (id) [inline src/brucke_metrics.erl]: same here

**Review by id — COMMENTED** [review]
Also please bump minor version in VSN

**Comment by id** [comment]
@DEvil0000 could you please resolve conflicts?

File: src/brucke_http.erl [hunk 3]
```diff
@@ -38,7 +38,7 @@ init() ->
         {error, {already_started, _Pid}} ->
           ok;
         _else ->
```
Comment (id) [inline src/brucke_http.erl]: This one should be `Other`, and then you propagate the error up

**Review by id — APPROVED** [review]
Thanks!

**Pull request merged**